samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -150,7 +150,10 @@ static int ble_adv_start(void)
 	err = bt_le_adv_start(adv_param, ad, ARRAY_SIZE(ad),
 			      sd, ARRAY_SIZE(sd));
 
-	if (err) {
+	if (err == -ECONNREFUSED) {
+		LOG_WRN("Already connected, do not advertise");
+		return 0;
```

Comment:
set state to IDLE as this can be called when state is STATE_GRACE_PERIOD

also instead of return use `err = 0` and allow function to reach common exit point

Reply:
> BTW - stop should not handle ECONNREFUSED too? (if so then set idle there)

Advertising stop returns success if there is no advertising to stop (like in the connected state) so we don't need to handle it additionally.

Reply:
> also instead of return use `err = 0` and allow function to reach common exit point

Fixed